modules/imgproc/test/test_drawing.cpp

```diff
@@ -680,6 +680,120 @@ TEST(Drawing, fillpoly_circle)
     EXPECT_LT(diff_fp3, 1.);
 }
 
+TEST(Drawing, fillpoly_contours)
+{
+    int imgSize = 50;
+    int type = CV_8UC1;
+    int shift = 0;
+    cv::LineTypes lineType = LINE_8;
+
+    cv::Mat img(imgSize, imgSize, type);
+
+    // check that contours of fillPoly and polylines match
+    {
+        std::vector<cv::Point> polygonPoints;
+        polygonPoints.push_back(cv::Point(44, 27));
+        polygonPoints.push_back(cv::Point(7, 37));
+        polygonPoints.push_back(cv::Point(7, 19));
+        polygonPoints.push_back(cv::Point(38, 19));
+
+        std::vector<const cv::Point*> polygonPointPointers(polygonPoints.size());
+        for (size_t i = 0; i < polygonPoints.size(); i++)
+        {
+            polygonPointPointers[i] = &polygonPoints[i];
+        }
+
+        const cv::Point** data = &polygonPointPointers.front();
+        int size = (int)polygonPoints.size();
+        const int* npts = &size;
+        int ncontours = 1;
+
+        img = 0;
+        cv::fillPoly(img, data, npts, ncontours, 255, lineType, shift);
+        cv::polylines(img, data, npts, ncontours, true, 0, 1, lineType, shift);
+        {
+            cv::Mat labelImage(img.size(), CV_32S);
+            int labels = cv::connectedComponents(img, labelImage, 4);
+            EXPECT_EQ(2, labels) << "filling went over the border";
+        }
+    }
+
+    // check that line generated with fillPoly and polylines match
+    {
+        std::vector<cv::Point> polygonPoints;
+        polygonPoints.push_back(cv::Point(44, 27));
+        polygonPoints.push_back(cv::Point(38, 19));
+
+        std::vector<const cv::Point*> polygonPointPointers(polygonPoints.size());
+        for (size_t i = 0; i < polygonPoints.size(); i++)
+        {
+            polygonPointPointers[i] = &polygonPoints[i];
+        }
+
+        const cv::Point** data = &polygonPointPointers.front();
+        int size = (int)polygonPoints.size();
+        const int* npts = &size;
+        int ncontours = 1;
+
+        img = 0;
+        cv::fillPoly(img, data, npts, ncontours, 255, lineType, shift);
+        cv::polylines(img, data, npts, ncontours, true, 0, 1, lineType, shift);
+        EXPECT_EQ(0, cv::countNonZero(img)) << "lines do not match";
```

Comment:
Instead of two comparison we can make one simpler check:
```.cpp
im1 = 0; img2 = 0;
cv::fillPoly(img1, ...);
cv::polylines(img2, ...); // same color
EXPECT_MAT_N_DIFF(img1, img2, 0);
```

Reply:
Adjusted it.